pkg/k8s/finalizers.go

```diff
@@ -38,34 +34,46 @@ type defaultFinalizerManager struct {
 	log       logr.Logger
 }
 
-func (m *defaultFinalizerManager) AddFinalizers(ctx context.Context, obj APIObject, finalizers ...string) error {
-	oldObj := obj.DeepCopyObject()
-	needsUpdate := false
-	for _, finalizer := range finalizers {
-		if !HasFinalizer(obj, finalizer) {
-			controllerutil.AddFinalizer(obj, finalizer)
-			needsUpdate = true
+func (m *defaultFinalizerManager) AddFinalizers(ctx context.Context, obj client.Object, finalizers ...string) error {
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
is there a unit-test to verify this new retry logic? I am assuming this is pretty standard snippet across other controller implementation. It may help to add a note on why retry is needed here. Thanks

Reply:
We do have finalizers_test. Will check if we need to update it to verify the retry logic